### PR TITLE
Create an empty centroids in Hazard.concat() with CRS from the first haz_list.

### PR DESCRIPTION
**Pull request by manniepmkam**
Changes proposed in this PR:

This pull request is a fix for issue #879. 

An empty centroids is created in the `Hazard.concat()` function with the crs from the haz object from the haz_list.

This PR fixes #

### PR Author Checklist

- [x] Read the [Contribution Guide][contrib]
- [x] Correct target branch selected (if unsure, select `develop`)
- [x] Descriptive pull request title added
- [x] Source branch up-to-date with target branch
- [ ] [Documentation](https://climada-python.readthedocs.io/en/latest/guide/Guide_PythonDos-n-Donts.html#2.--Commenting-&-Documenting) updated
- [x] [Tests][testing] updated
- [x] [Tests][testing] passing
- [ ] No new [linter issues][linter]
- [ ] [Changelog](https://github.com/CLIMADA-project/climada_python/blob/main/CHANGELOG.md) updated

### PR Reviewer Checklist

- [ ] Read the [Contribution Guide][contrib]
- [ ] [CLIMADA Reviewer Checklist](https://climada-python.readthedocs.io/en/latest/guide/Guide_Reviewer_Checklist.html) passed
- [ ] [Tests][testing] passing
- [ ] No new [linter issues][linter]

[contrib]: https://github.com/CLIMADA-project/climada_python/blob/main/CONTRIBUTING.md
[testing]: https://climada-python.readthedocs.io/en/latest/guide/Guide_Continuous_Integration_and_Testing.html
[linter]: https://climada-python.readthedocs.io/en/stable/guide/Guide_Continuous_Integration_and_Testing.html#3.C.--Static-Code-Analysis

### CODE REVIEW

File: climada/hazard/base.py [hunk 1]
```diff
@@ -818,6 +818,7 @@ def concat(cls, haz_list):
             return cls()
         haz_concat = haz_list[0].__class__()
         haz_concat.haz_type = haz_list[0].haz_type
+        haz_concat.centroids = Centroids(lat=[], lon=[], crs=haz_list[0].centroids.crs)
```
Comment (emanuel-schmid) [inline climada/hazard/base.py:821]: Exactly! 😄 
Alternatively one may inject the centroids as an argument of `haz_list[0].__class__`. (And possibly skip the `haz_type` assignment. I suspect that it is already assigned at that point through the constructor)

**Comment by emanuel-schmid** [comment]
Very well! Can you add tests and changelog?

**Review by emanuel-schmid — CHANGES_REQUESTED** [review]

**Review by peanutfun — CHANGES_REQUESTED** [review]
@manniepmkam, as discussed in person, please
* fix the linter issue: Line too long
* add a unit test that covers the case reported in https://github.com/CLIMADA-project/climada_python/issues/879#issuecomment-2096281682

**Comment by manniepmkam** [comment]
> @manniepmkam, as discussed in person, please
> 
> * fix the linter issue: Line too long
I have updated the line.

> * add a unit test that covers the case reported in [Issues with CRS when using  TropCyclone.from_tracks() #879 (comment)](https://github.com/CLIMADA-project/climada_python/issues/879#issuecomment-2096281682)
The unit test for hazard.base has been updated accordingly. Please review that.

File: climada/hazard/test/test_base.py [hunk 2]
```diff
@@ -900,15 +904,20 @@ def test_append_raise_type_error(self):
...
         haz3 = Hazard('EQ')
         with self.assertRaises(ValueError):
             Hazard.concat([haz1, haz3])
 
         haz4 = Hazard('TC', units='cm')
         with self.assertRaises(ValueError):
             Hazard.concat([haz1, haz4])
+            
+        haz5 = Hazard('TC', centroids=Centroids(lat=[],lon=[], crs="epsg:7777"))
+        with self.assertRaises(ValueError):
```
Comment (peanutfun) [inline climada/hazard/test/test_base.py:919]: Please check the exact error that is raised using [`self.assertRaisesRegex`](https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertRaisesRegex), where the second argument is an expression that should appear in the error message. Notice that `haz5` does not define `units`, so the error might also originate from that.

**Review by peanutfun — CHANGES_REQUESTED** [review]
Looks good, but you should check for the exact error message and not just any `ValueError`, see my comment.

**Review by peanutfun — APPROVED** [review]

**Comment by peanutfun** [comment]
@manniepmkam Thanks a lot for the bugfix! 🙌

**Pull request merged**